main.cpp

```diff
@@ -0,0 +1,54 @@
+#include "units.h"
+#include <iostream>
+
+void listUnitAttributes(Unit& unit1,Unit& unit2){	
+	std::cout << unit1.getName() << ":HP:" << unit1.getHp() << ",DMG:" << unit1.getDamage() << std::endl;
+	std::cout << unit2.getName() << ":HP:" <<unit2.getHp() << ",DMG:" <<unit2.getDamage() << std::endl;
+}
+void damageLog(Unit& unit1, Unit& unit2) {
+		unit1.dealDamageTo(unit2);
+		std::cout << unit1.getName() << "->" << unit2.getName() << std::endl;
+}
+
+void gamePlay(Unit& unit1, Unit& unit2){
+	listUnitAttributes(unit1, unit2);
+	bool gameEnd=false;
+	while(!gameEnd){
+		//Az első unit támad
+		if(unit1.getHp() != 0)
+		{
+			damageLog(unit1, unit2);
+			listUnitAttributes(unit1, unit2);
+		}
+		else{
+			std::cout << unit1.getName() << " died." << unit2.getName() << " wins." << std::endl;
+			gameEnd = true;
+			break;
+		}
+		//A második unit támad
+		if (unit2.getHp() != 0)
+		{
+			damageLog(unit2, unit1);
+			listUnitAttributes(unit1, unit2);
+		}
+		else
+		{
+			std::cout << unit2.getName() << " died." << unit1.getName() << " wins." << std::endl;
+			gameEnd = true;
+		}

```

Comment:
Ezt a ciklusfeltételt és az if szerkezetet gyúrjátok újra. Nem vagyok benne biztos hogy ez a legjobb megoldás.

Reply:
Esetleg rekurzioval meglehetne oldani?

Reply:
Minden cilkus kiváltható rekurzióval, de ne. Rendben van az hogy ciklus. A gamEnd-re gondoltam, hogy az talán valahogy beépíthető a feltételbe.

Reply:
Rendben van, koszonjuk!

Reply:
A gameEnd változót kitöröltük, a feltételt átírtuk.